components/TARGET_PSA/services/psa_prot_internal_storage/COMPONENT_PSA_SRV_IMPL/pits_impl.cpp

```diff
@@ -133,19 +188,20 @@ static void generate_fn(char *tdb_filename, uint32_t tdb_filename_size, uint32_t
     // Iterate on UID; each time convert 6 bits of UID into a character; first iteration must be done
     do {
         tdb_filename[filename_idx++] = base64_coding_table[uid & 0x3F];
-        uid = lsr(uid, 6);
+        uid = lsr64(uid, 6);
     } while (uid != 0);
 
     tdb_filename[filename_idx++] = '\0';
     MBED_ASSERT(filename_idx <= PSA_ITS_FILENAME_MAX_LEN);
 }
 
-psa_its_status_t psa_its_set_impl(int32_t pid, uint32_t uid, uint32_t data_length, const void *p_data, psa_its_create_flags_t create_flags)
+psa_its_status_t psa_its_set_impl(int32_t pid, psa_its_uid_t uid, uint32_t data_length, const void *p_data, psa_its_create_flags_t create_flags)
 {
-    KVStore *kvstore = get_kvstore_instance();
-    MBED_ASSERT(kvstore);
+    if (!kvstore) {
+        its_init();
+    }
 
-    if ((create_flags != 0) && (create_flags != PSA_ITS_WRITE_ONCE_FLAG)) {
+    if ((create_flags & (~PSA_ITS_FLAGS_BIT_MASK)) != 0) {
```

Comment:
`PSA_ITS_FLAGS_BIT_MASK` is your new addition to the spec - (https://github.com/ARMmbed/psa_trusted_storage_api/pull/22) which is not approved yet.
i would suggest not using it until it is accepted.